preprocess.py

```diff
@@ -9,8 +10,53 @@ def read_text(filename: str) -> str:
 
         return file_string
 
+# Helper functions for preproc_text()
+def __remove_new_line(string: str) -> str:
+    """
+    Return a copy of the string with new lines replaced with a single space.
+    """
+    return string.replace("\n", " ")
 
+def __remove_contractions(string: str) -> str:
+    """
+    Return a copy of the string with English contractions replaced.
+    WARNING: will change word count.
+    """
+    string = re.sub(r'\'s', ' is', string) 
+    string = re.sub(r'n\'t', ' not', string)
+    string = re.sub(r'\'ve', ' have', string)
+    string = re.sub(r'\'re', ' are', string)
+    string = re.sub(r'\'m', ' am', string)
+    string = re.sub(r'\'ll', ' will', string)
```

Comment:
this could probably be combined into one line using regex, but I think for clarity, it's ok to not do so!

Reply:
Yeah, this code is very readable.